Hello,

On Apple hardware when running pyinstaller:

I didn't explicitly set TK_LBIRARY or TK_LIBRARY

I do have Tk installed via pip.

I found this in the pyinstaller build:

1949 INFO: TclTkInfo: initializing cached Tcl/Tk info...
1971 WARNING: tkinter installation is broken. It will be excluded from the application

My built application errors out with:

Traceback (most recent call last):
  File "eisenban.py", line 9, in <module>
    from tkinter import Tk, messagebox
ModuleNotFoundError: No module named 'tkinter'
[PYI-19709:ERROR] Failed to execute script 'eisenban' due to unhandled exception: No module named 'tkinter'
[PYI-19709:ERROR] Traceback:
Traceback (most recent call last):
  File "eisenban.py", line 9, in <module>
    from tkinter import Tk, messagebox
ModuleNotFoundError: No module named 'tkinter'

Let me know if you need more information.

--- a/PyInstaller/utils/hooks/tcl_tk.py
+++ b/PyInstaller/utils/hooks/tcl_tk.py
@@ -169,10 +169,13 @@ class TclTkInfo:
         # in the isolated subprocess as part of `_get_tcl_tk_info`. However, that is impractical, as it shows the empty
         # window, and on some platforms (e.g., linux) requires display server. Therefore, try to guess the location,
         # based on the following heuristic:
+        #  - if TK_LIBRARY is defined use it.
         #  - if Tk is built as macOS framework bundle, look for Scripts sub-directory in Resources directory next to
         #    the shared library.
         #  - otherwise, look for: $tcl_root/../tkX.Y, where X and Y are Tk major and minor version.
-        if compat.is_darwin and self.tk_shared_library and (
+        if "TK_LIBRARY" in os.environ:
+            self.tk_data_dir = os.environ["TK_LIBRARY"]
+        elif compat.is_darwin and self.tk_shared_library and (
             # is_framework_bundle_lib handles only fully-versioned framework library paths...
             (osxutils.is_framework_bundle_lib(self.tk_shared_library)) or
             # ... so manually handle top-level-symlinked variant for now.

